src/web.ts

```diff
@@ -1,22 +1,86 @@
-import { WebPlugin } from '@capacitor/core';
-import { SignInWithApplePlugin } from './definitions';
+import { WebPlugin } from "@capacitor/core";
+
+import {
+  SignInWithApplePlugin,
+  InitOptions,
+  SignInResponse,
+  SignInError,
+} from "./definitions";
+
+declare var window: any;
+
+export class SignInWithAppleWeb extends WebPlugin
+  implements SignInWithApplePlugin {
+  hasInitialized = false;
 
-export class SignInWithAppleWeb extends WebPlugin implements SignInWithApplePlugin {
   constructor() {
     super({
-      name: 'SignInWithApple',
-      platforms: ['web']
+      name: "SignInWithApple",
+      platforms: ["web"],
+    });
+  }
+
+  async Init(options: InitOptions): Promise<void> {
+    this.loadAppleScript(() => {
+      if (window && window.AppleID && !this.hasInitialized) {
+        const { clientId, scope, redirectURI, state, usePopup } = options;
+
+        window.AppleID.auth.init({
+          clientId: clientId,
+          scope: scope,
+          redirectURI: redirectURI,
+          state: state,
+          usePopup: usePopup !== undefined ? usePopup : false,
+        });
+
+        this.hasInitialized = true;
+      }
+    });
+  }
+
+  Authorize(): Promise<SignInResponse> {
+    return new Promise<SignInResponse>(async (resolve, reject) => {
+      try {
+        if (window && window.AppleID) {
+          if (this.hasInitialized) {
+            const response: SignInResponse = await window.AppleID.auth.signIn();
+            resolve(response);
+          } else {
+            reject({
+              error: "AppleID has not yet initialized",
+            } as SignInError);
+          }
+        } else {
+          reject({
+            error: "Cannot find AppleID instance",
+          } as SignInError);
+        }
+      } catch (error) {
+        reject(error as SignInError);
+      }
```

Comment:
this covers the scenarios, but it's not that readable, you could flatten this out quite a lot?

Reply:
Addressed in ea9e78d